Sources/Fuzzilli/Fuzzer.swift

```diff
@@ -426,16 +467,47 @@ public class Fuzzer {
     ///   - program: The FuzzIL program to execute.
     ///   - timeout: The timeout after which to abort execution. If nil, the default timeout of this fuzzer will be used.
     /// - Returns: An Execution structure representing the execution outcome.
-    public func execute(_ program: Program, withTimeout timeout: UInt32? = nil) -> Execution {
+    public func execute(_ program: Program, withTimeout timeout: UInt32? = nil,
+                        differentialExecute: DifferentialExecute = .auto) -> Execution {
         dispatchPrecondition(condition: .onQueue(queue))
         assert(runner.isInitialized)
 
-        let script = lifter.lift(program)
+        let differentialResult: Bool
+        switch differentialExecute {
+            case .force:
+                differentialResult = true
+            case .disabled:
+                differentialResult = false
+            case .auto:
+                differentialResult = program.code.contains { $0.isDifferentialHash }
+        }
 
+        let script = lifter.lift(program)
         dispatchEvent(events.PreExecute, data: program)
-        let execution = runner.run(script, withTimeout: timeout ?? config.timeout)
+        var execution = runner.run(script, withTimeout: timeout ?? config.timeout)
         dispatchEvent(events.PostExecute, data: execution)
 
+        if differentialResult && shouldExecutionBeIgnoredForDifferentialTesting(execution: execution) {
+            // deliberate crash by the engine due to a benign, non-deterministic execution
+            execution.outcome = .failed(0)
+        }
+        else if differentialResult && execution.outcome == .succeeded &&
+           !shouldExecutionBeIgnoredForDifferentialTesting(execution: execution) {
+
+            assert(referenceRunner != nil)
+            assert(referenceRunner!.isInitialized)
+            let diff = referenceRunner!.run(script, withTimeout: timeout ?? config.timeout)
+            dispatchEvent(events.PostExecute, data: diff)
```

Comment:
The reason I added dispatching of `PostExecute` is to increase the `totalExecs`, buts its not really necessary. Omitting the event would result in `totalExecs` not including the executions in the reference runner.
Dispatching `PreExecute` is unnecessary work at this would cause a call to `cov_clear_bitmap`.